packages/ui/components/list/List.tsx

```diff
@@ -1,129 +1,142 @@
+      <>
+        <Wrapper>
+          <div className="flex [&>*]:mr-3">
+            {leftNode ? <div className="max-w-9 max-h-9">{leftNode} </div> : null}
+            <div className="flex flex-col">
+              <BadgeHandler
+                as="h3"
+                className="h-5 text-sm font-semibold leading-none text-black"
+                componentPosition="heading"
+                badges={props.badges}
+                badgePosition={props.badgePosition}>
+                {heading}
+              </BadgeHandler>
+              {subHeading && (
+                <BadgeHandler
+                  as="h4"
+                  className="text-sm font-normal leading-normal text-gray-600"
+                  componentPosition="subheading"
+                  badges={props.badges}
+                  badgePosition={props.badgePosition}>
+                  <div className="line-clamp-1">{subHeading}</div>
+                </BadgeHandler>
+              )}
+              <BadgeHandler
+                componentPosition="below"
+                badges={props.badges}
+                badgePosition={props.badgePosition}
+              />
+              {props.belowHeading ? props.belowHeading : null}
+            </div>
+            {actions ? <div className="ml-auto">{actions}</div> : null}
+          </div>
+        </Wrapper>
+        {expanded ? (
+          <>
+            <div className="my-4 h-px bg-gray-200" />
+            <div ref={animatedRef}>{expanded}</div>
+          </>
+        ) : null}
+      </>
     </li>
   );
 }
 
-export function ListItemTitle<TComponent extends keyof JSX.IntrinsicElements = "span">(
-  props: JSX.IntrinsicElements[TComponent] & { component?: TComponent }
-) {
-  const { component = "span", ...passThroughProps } = props;
+type BadgeHandlerProps = {
+  badgePosition?: "heading" | "subheading" | "below";
+  badges?: ReactNode;
+  componentPosition?: "heading" | "subheading" | "below";
+  as?: React.ElementType;
+  className?: string;
+};
 
-  return createElement(
-    component,
-    {
-      ...passThroughProps,
-      className: classNames("text-sm font-medium text-gray-900 truncate", props.className),
-    },
-    props.children
-  );
-}
+function BadgeHandler(props: PropsWithChildren<BadgeHandlerProps>) {
+  const Component = props.as || "div";
 
-export function ListItemText<TComponent extends keyof JSX.IntrinsicElements = "span">(
-  props: JSX.IntrinsicElements[TComponent] & { component?: TComponent }
-) {
-  const { component = "span", ...passThroughProps } = props;
+  if (props.badgePosition === props.componentPosition) {
+    return (
+      <div className="item-center flex">
+        <Component className={classNames(props.className, "inline-flex items-center")}>
+          {props.children}
+        </Component>
+        <div
+          className={classNames("flex", props.badgePosition === "below" ? "[&>*]:mr-2" : "ml-1 space-x-2")}>
```

Comment:
When its below we dont want ml applied so we use child selectors to apply mr to all children (emulates gap)

Reply:
When explaining this is realised gap would probably work instead of this. no biggy tho